app/main.py

```diff
@@ -1,2 +1,46 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.storage = [None] * self.capacity
+
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.next = None
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        index = hash(key) % self.capacity
+        node = self.storage[index]
+
+        while node:
+            if node.key == key:
+                node.value = value
+                return
+            node = node.next
+
+        new_node = self.Node(key, value)
+        new_node.next = self.storage[index]
+        self.storage[index] = new_node
+        self.size += 1
```

Comment:
So, your storage will only have 8 capacity? What about resizing of this dictionary?

>Load Factor: When should your dictionary resize to maintain efficiency?
>Resize Strategy: How should your dictionary expand its capacity when needed?